File: php/grade-school/grade-school.php
<?php

class School {

    protected $school;

    public function __construct(){

        $this->school = [];
        
    }

    public function add($name, $grade){
        if (key_exists($grade, $this->school)) {
            $this->school[$grade] = sort(array_merge([$name], $this->school[$grade]));
        } else {
            $this->school[$grade] = [$name];
        }
    }

    public function grade($grade){
        
        
    }
    public function getSchool(){

        return $this->school;
        
    
    }
}

File: ruby/grade-school/grade_school.rb
class School
  def initialize
    @school = {}
  end

  def add(student, grade)
    @school.store(student, grade)
  end

  def students(grade)
    match = ->(_k, v) { v == grade }
    @school.select(&match).keys.sort
  end

  def students_by_grade
    formated_school = @school.each_with_object(Hash.new([])) do |(student, grade), memo|
      memo[grade] += student.split
    end
    formated_school.sort.map { |k, v| { grade: k, students: v.sort } }
  end
end

module BookKeeping
  VERSION = 3
end
